tests/run/xml-interpolation/XmlQuote_1.scala

```diff
@@ -0,0 +1,52 @@
+import scala.quoted._
+import scala.tasty.Tasty
+
+case class Xml(parts: String, arg: Any)
+
+class XmlQuote(ctx: StringContext) {
+  rewrite def xml(args: => Any*): Xml = ~XmlQuote.impl('(ctx), '(args))
+}
+
+object XmlQuote {
+  implicit rewrite def XmlQuote(ctx: => StringContext): XmlQuote = new XmlQuote(ctx)
+
+  def impl(ctx: Expr[StringContext], args: Expr[Seq[Any]])(implicit tasty: Tasty): Expr[Xml] = {
+    import tasty._
+    import Term._
+
+    // for debugging purpose
+    def pp(tree: Tree): Unit = {
+      println(tasty.showSourceCode.showTree(tree))
+    }
+
+    def isStringConstant(tree: Term) = tree match {
+      case Literal(_) => true
+      case _ => false
+    }
+
+    // _root_.scala.StringContext.apply(("p0", "p1": scala.<repeated>[scala#Predef.String]))
+    val parts = ctx.toTasty match {
+      case Inlined(_, _,
+        Apply(
+          Select(Select(Select(Ident("_root_"), "scala", _), "StringContext", _), "apply", _),
```

Comment:
In `tasty.definition` you should have types or symbols for root and scala. You should be able to test this in the guard.

Reply:
Doesn't look simpler. It is more correct?

Reply:
Then this is fine. I expected it to look simpler